junit5/src/main/java/com/linecorp/armeria/testing/junit5/client/TestHttpResponse.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.junit5.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.ResponseHeaders;
+
+/**
+ * A complete HTTP response includes fluent test methods. whose content is readily available as a single {@link HttpData}.
+ */
+public interface TestHttpResponse extends AggregatedHttpMessage {
+
+    static TestHttpResponse of(AggregatedHttpResponse aggregatedHttpResponse) {
```

Comment:
In addition to a normal response, a client throws an exception or an `HttpResponse` completes exceptionally. Should we add more methods to assert exceptions?

```java
server
  .webTestClient()
  .execute(req)
  .assertCause()
  .isInstanceOf(UnprocessedRequestException.class)
  // The assertion for cause does not have to back `TestHttpResponse`
  // because there are no other elements to assert.
  .hasMessage("....")
```

Reply:
I think it's necessary to have an API. 👍 I will add the methods.